server/src/main/java/io/crate/analyze/ShowStatementAnalyzer.java

```diff
@@ -87,12 +89,17 @@ AnalyzedStatement analyzeShowTransaction(Analysis analysis) {
     }
 
     public AnalyzedStatement analyzeShowCreateTable(Table<?> table, Analysis analysis) {
-        DocTableInfo tableInfo = (DocTableInfo) schemas.resolveTableInfo(
+        CoordinatorSessionSettings sessionSettings = analysis.sessionSettings();
+        RelationInfo relation = schemas.resolveRelationInfo(
             table.getName(),
             Operation.SHOW_CREATE,
-            analysis.sessionSettings().sessionUser(),
-            analysis.sessionSettings().searchPath()
+            sessionSettings.sessionUser(),
+            sessionSettings.searchPath()
         );
+        if (!(relation instanceof TableInfo tableInfo)) {
+            throw new UnsupportedOperationException(
+                "Cannot use SHOW CREATE TABLE on relation " + relation.ident() + " of type " + relation.relationType());
+        }
```

Comment:
I think this is not needed because `schemas.resolveRelationInfo` would throw `OperationOnInaccessibleRelationException` in such cases.

Reply:
It actually uses the cast result (`tableInfo`) because `AnalyzedShowCreateTable` is stricter typed
I could replace it with an assert, and regular cast, but I don't really see the advantage of that

And the contract that Operation.SHOW_CREATE -> returns DocTableInfo or ForeignTable is rather loose.

Reply:
Thank you, I see now.